pkg/provisioner/cloudprovisioner.go

```diff
@@ -329,36 +329,39 @@ func (c *CloudProvisioner) ListVolumes(
 	return c.listVolumesInNodeResourceGroup(ctx, start, int(maxEntries))
 }
 
+// PublishVolume calls AttachDisk asynchronously and returns early lun assignment value and a channel for the async attach results.
 func (c *CloudProvisioner) PublishVolume(
 	ctx context.Context,
 	volumeID string,
 	nodeID string,
-	volumeContext map[string]string) (map[string]string, error) {
+	volumeContext map[string]string) (chan azureutils.CloudAttachResult, map[string]string, error) {
```

Comment:
Rather than return the `PublishVolume` result either through the channel or by direct return (forcing the caller to handle both cases), it would be cleaner and simpler to *always* return results through the channel.